src/Search/Console.php

```diff
@@ -119,6 +122,75 @@ private function enqueue($type, $id)
         $this->logger->info('Item added successfully.');
     }
 
+    public function indexRebuildCommand(){
+
+        $newIndexName = 'elife_search_tmp';
+        $client = $this->getElasticClient();
+
+        // Delete the existing 'elife_search_tmp' index if we have one
+        echo "checking if an old 'elife_search_tmp' index exists \n";
+        $response = $client->deleteIndex($newIndexName);
+
+        if ($response['payload'] instanceof \eLife\Search\Api\Elasticsearch\Response\SuccessResponse){
+
+            echo "Found an old 'elife_search_tmp' index, Deleted it \n";
+
+        } else if ($response['payload'] instanceof \eLife\Search\Api\Elasticsearch\Response\ErrorResponse){
+
+            if ($response['payload']->error['type'] == 'index_not_found_exception'){
+
+                echo "Did not find an old 'elife_search_tmp' index, This is okay .. continuing \n";
+
+            }else{
+
+                echo "Something went wrong, we got an exception we were not expecting \n";
+
+            }
+        }
+
+        echo "Creating a new (empty) 'elife_search_tmp' index  \n";
+        $response = $client->createIndex($newIndexName);
+
+        if ($response['payload'] instanceof \eLife\Search\Api\Elasticsearch\Response\SuccessResponse){
+
+            echo "Successfully created the new index \n";
+
+        }else {
+
+            echo "Could not create the new index ... exiting \n";
+            die();
+
+        }
+
+        // Kill all existing Gearnman workers
+        // Giorgio perhaps some linux command
+
+        // start some new ones using our new index
+        // Populate exiting items
+        //subsribe to SQS queue
+        exec('./bin/console gearman:worker  --index="elife_search_tmp" >> /tmp/gearman-worker.log 2>&1 &');
```

Comment:
I think this section has to be managed outside of this script. We could make the assumption that nothing is running here, and just populate the new index without touching the existing one. From the deployment, we will be able to:
1. stop all workers
1. call `indexRebuildCommand()`
1. switch the index
1. start all workers again

During this process, the application will always be able to serve reads:
- the old, stale index
- the new, stale index as it catches up (after the switch)
- the new, up-to-date index (a few seconds/minutes after the switch)

So I think this command should be limited to: build a new index with a custom name.
Then we have the problem of switching, but I would separate that problem into another command as we have to decide:
- how to store the index name (some config file)
- how to switch (rewrite the file?)